app/controllers/v1/AdminDataController.scala

```diff
@@ -0,0 +1,83 @@
+package controllers.v1
+
+/**
+ * Created by coolit on 16/11/2017.
+ */
+import javax.inject.Inject
+
+import scala.concurrent.Future
+import scala.util.{ Failure, Success, Try }
+
+import akka.pattern.ask
+import play.api.cache.CacheApi
+import play.api.i18n.{ I18nSupport, Messages, MessagesApi }
+import play.api.libs.json.Json
+import play.api.mvc.{ Action, AnyContent, Result }
+import org.joda.time.YearMonth
+import org.joda.time.format.DateTimeFormat
+import com.typesafe.scalalogging.LazyLogging
+
+import config.Properties._
+import models.ValidLookup
+import hbase.model.AdminData
+import hbase.repository.AdminDataRepository
+import utils.LookupValidator.REFERENCE_PERIOD_FORMAT
+import utils.{ LookupValidator, Utilities }
+
+/**
+ * Created by coolit on 07/11/2017.
+ */
+class AdminDataController @Inject() (repository: AdminDataRepository, cache: CacheApi, val messagesApi: MessagesApi) extends ControllerUtils with I18nSupport with LazyLogging {
+
+  val cb = getCircuitBreaker(getRecordById)
+
+  def lookup(period: Option[String], id: String): Action[AnyContent] = Action.async { implicit request =>
+    logger.info(s"Lookup with period [$period] for id [$id]")
+    LookupValidator.validateLookupParams(id, period) match {
+      case Right(v) => {
+        val cacheKey = List(v.id, v.period.getOrElse(None)).mkString(cacheDelimiter)
+        cache.get[Future[Result]](cacheKey).getOrElse(repositoryLookup(v, cacheKey))
+      }
+      case Left(error) => BadRequest(Utilities.errAsJson(BAD_REQUEST, "Bad Request", error.msg)).future
```

Comment:
Don't hard-code error messages that are returned by the API, use the resource bundle conf/messages
Multiple instances of hard-coding within this class